### ISSUE TITLE
Bulk Index timeout

### ISSUE THREAD

**Issue by alkaros2506** [elastic/elasticsearch-py#517]
passing `params=dict(request_timeout=100)` as
```
.bulk(..., params=dict(request_timeout=100))
```
works but when doing chunks it doesn't
```
.bulk(..., chunk_size=100, max_chunk_bytes=100000, params=dict(request_timeout=100))
```
because `params` pass to `elasticsearch.helpers.streaming_bulk` as `**kwargs` are then passed all the way down to `elasticsearch.transport.Transport#perform_request` and `kwargs` gets mutated by `timeout = params.pop('request_timeout', None)` which results in subsequents requests to be missing some parameters.

**Comment by alkaros2506** [comment]
@HonzaKral Can I ask when you're planning on releasing this fix? Will it be available in the 2.4.* versions?

**Comment by HonzaKral** [comment]
Sorry for the delay, this has been released as part of the `5.2.0` release.
